include/ros1_bridge/factory.hpp

```diff
@@ -145,10 +149,22 @@ class Factory : public FactoryInterface
   static
   void ros2_callback(
     typename ROS2_T::SharedPtr ros2_msg,
+    const rmw_message_info_t & msg_info,
     ros::Publisher ros1_pub,
     const std::string & ros1_type_name,
-    const std::string & ros2_type_name)
+    const std::string & ros2_type_name,
+    rclcpp::PublisherBase::SharedPtr ros2_pub = nullptr)
   {
+    if (ros2_pub) {
+      bool result = false;
+      auto ret = rmw_compare_gids_equal(&msg_info.publisher_gid, &ros2_pub->get_gid(), &result);
+      if (ret == RMW_RET_OK) {
+        if (result) {  // message GID equals to bridge's ROS2 publisher GID
+          return;  // do not publish messages from bridge itself
+        }
+      }
```

Comment:
The else case should be handled here. If `rmw_compare_gids_equal` fails, then we should probably raise an exception or at least log an error message to the console using `RCLCPP_ERROR` logging macro.

Reply:
You can get the underlying error message with `rmw_get_error_string_safe()`, and after getting it you should clear the error state with `rmw_reset_error()`.

Reply:
Ok, I have found similar code on different repo. Added.